src/main/kotlin/com/terraformation/backend/accelerator/model/SpeciesDeliverableSubmissionModel.kt

```diff
@@ -0,0 +1,26 @@
+package com.terraformation.backend.accelerator.model
+
+import com.terraformation.backend.db.accelerator.DeliverableId
+import com.terraformation.backend.db.accelerator.SubmissionId
+import com.terraformation.backend.db.accelerator.tables.references.DELIVERABLES
+import com.terraformation.backend.db.accelerator.tables.references.SUBMISSIONS
+import org.jooq.Record
+
+data class ExistingSpeciesDeliverableSubmissionModel(
+    val deliverableId: DeliverableId,
+    val submissionId: SubmissionId? = null
+) {
+  companion object {
+    fun of(record: Record): ExistingSpeciesDeliverableSubmissionModel {
+      if (record[SUBMISSIONS.ID] == null) {
+        return ExistingSpeciesDeliverableSubmissionModel(
+            deliverableId = record[DELIVERABLES.ID]!!, submissionId = null)
+      }
```

Comment:
Removing the `!!` from the submission ID value in the other `return` statement would do this too.